src/scripts.js

```diff
@@ -221,11 +216,30 @@ function openRecipeInfo(event) {
   domUpdates.makeInline(fullRecipeInfo);
   let recipeId = event.path.find(e => e.id).id;
   let recipe = recipeData.find(recipe => recipe.id === Number(recipeId));
+  console.log("RECIP", recipe)
 
-  domUpdates.generateRecipeTitle(recipe, generateIngredients(recipe, ingredientsData), fullRecipeInfo);
+  domUpdates.generateRecipeTitle(recipe, generateIngredients(recipe, ingredientsData), fullRecipeInfo, ingredientsData);
   domUpdates.addRecipeImage(recipe);
   domUpdates.generateInstructions(recipe, fullRecipeInfo);
   domUpdates.addOverlay(fullRecipeInfo);
+  let userShoppingList = document.querySelector('.recipe-instructions')
+  let htmlString = "<button class='cook-me'>Cook Me</button>"
+  userShoppingList.innerHTML += htmlString
+  let cookMeBtn = document.querySelector('.cook-me')
+  cookMeBtn.addEventListener('click', function() {
+    console.log("MATTH", recipe)
+    checkPantryIngredients(recipe, user.pantry)
+  })
+}
+
+function checkPantryIngredients(recipe, pantry) {
+  let missingIngredients = pantry.hasEnoughIngredients(recipe)
+  let userShoppingList = document.querySelector('.recipe-instructions')
+  missingIngredients.forEach(ingredient => {
+    console.log("KJ", ingredient)
+    let ingredientName = ingredientsData.find(i => i.id == Object.keys(ingredient)[0]).name
```

Comment:
UGH This is hideous. Let me rework the Pantry.js class so that the return data is easier to use. Sorry everyone!